src/pkg/variables/types.go

```diff
@@ -21,24 +22,31 @@ const (
 	FileVariableType VariableType = "file"
 )
 
-var (
-	// IsUppercaseNumberUnderscore is a regex for uppercase, numbers and underscores.
-	// https://regex101.com/r/tfsEuZ/1
-	IsUppercaseNumberUnderscore = regexp.MustCompile(`^[A-Z0-9_]+$`).MatchString
-)
+// UppercaseNumberUnderscorePattern is a regex for uppercase, numbers and underscores.
+// https://regex101.com/r/tfsEuZ/1
+const UppercaseNumberUnderscorePattern = `^[A-Z0-9_]+$`
 
 // Variable represents a variable that has a value set programmatically
 type Variable struct {
 	// The name to be used for the variable
-	Name string `json:"name" jsonschema:"pattern=^[A-Z0-9_]+$"`
+	Name string `json:"name"`
 	// Whether to mark this variable as sensitive to not print it in the log
 	Sensitive bool `json:"sensitive,omitempty"`
 	// Whether to automatically indent the variable's value (if multiline) when templating. Based on the number of chars before the start of ###ZARF_VAR_.
 	AutoIndent bool `json:"autoIndent,omitempty"`
 	// An optional regex pattern that a variable value must match before a package deployment can continue.
 	Pattern string `json:"pattern,omitempty"`
 	// Changes the handling of a variable to load contents differently (i.e. from a file rather than as a raw variable - templated files should be kept below 1 MiB)
-	Type VariableType `json:"type,omitempty" jsonschema:"enum=raw,enum=file"`
+	Type VariableType `json:"type,omitempty"`
+}
+
+// JSONSchemaExtend extends the generated json schema during `zarf internal gen-config-schema`
+func (Variable) JSONSchemaExtend(schema *jsonschema.Schema) {
+	kind, _ := schema.Properties.Get("type")
```

Comment:
Is this ignoring an error :thinking:

Reply:
It's ignoring the present flag but I'd rather have the code panic if it doesn't find something since this is only run on `zarf internal gen-config-schema`

Reply:
I can't change the signature to return an error, since this signature is necessary for this function to get called by the jsonschema package during schema creation

Reply:
We cannot be a production grade tool that panics.

Reply:
This function is only called during schema generation, `zarf internal gen-config-schema`, and this would only panic if, during development, someone changed the json tag to not equal what is in the `schema.Properties.Get` call. 

Given that we don't have control over this function signature I think there's two options: 
1. return when the key isn't found - IMO this is worse because there won't be any signal from the CLI that something went wrong during schema generation. I'm fine with this though, it would happen rarely & it should be relatively obvious from the git diff that something went wrong. 
2. Keep panicing, maybe explicitly panic with a clear error message.